### ISSUE TITLE
OData Unbound Action parameters always null since migration to .NET Core 3.1

### ISSUE THREAD

**Issue by ben5Kuda** [microsoft/aspnet-api-versioning#636]
I have noticed that ODataActionParameters are always null since migrating to .NET Core 3.1. This is only affecting unbound actions. See sample below.

```
    [ODataRoute("CreateUser")]
    public IActionResult CreateUser(ODataActionParameters parameters)
```
See below for the request body

```
{
  "user": {
    "name": "Jon",
    "profile": "Admin
  }
}
```

If i inspect the raw HTTP request i see the following under QueryString `?user=name,Jon,profile,Admin`

**OData config**     
```
   var addUser = builder.Action("CreateUser");
   addUser.Parameter<User>("user").Required();
```
       public class User
       {
  	    public string Name { get; set;}
	    public string Profile { get; set;}
       }

For some reason, the parameters at method level are always null. I am not sure what has changed since .NET Core 2.2

I am currently using the following packages

```
 
<PackageReference Include="Microsoft.AspNetCore.OData" Version="7.4.0-beta" />
    
<PackageReference Include="Microsoft.AspNetCore.OData.Versioning" Version="4.1.1" />
    
<PackageReference Include="Microsoft.AspNetCore.OData.Versioning.ApiExplorer" Version="4.1.1" />
 
 <PackageReference Include="Swashbuckle.AspNetCore" Version="5.2.1" />

```

**Comment by commonsensesoftware** [comment]
Checking in. Were you ever able to resolve this?

This _looks_ like an OData misconfiguration, but it's hard to be sure. The parameter `user` is not an intrinsic type, so I would expect it to have be mapped as a _Complex Type_ (which I don't see). OData can be _lossey-goosey_ with its conventions. An _OData Action_ **must** be a `POST`. If memory serves me correct, the mapped controller method must either start with `Post` or be decorated with `[HttpPost]`.

If you can provide some additional information, we can dig into it deeper.

**Comment by commonsensesoftware** [comment]
A quick follow up on this. I see you mentioned values in the query string. This is almost certainly because the action is not mapped as `POST`. An OData action **must** be a `POST` and the parameters **must** be specified in the body (e.g. not the query string).

**Comment by commonsensesoftware** [comment]
Based on the original details, it appears this was using an early version of OData on .NET Core 3.1. This issue may be related to the attributes I specified or a bug in OData. It would seem that this issue has been resolved or abandoned. [5.0.0-RC.1](https://www.nuget.org/packages/Microsoft.AspNetCore.OData.Versioning/5.0.0-rc.1) is now available with support for _Endpoint Routing_. I am not able to reproduce this issue. The official release will likely occur after 2 weeks of _burn-in_. Thanks.
